# mycobiome_bacteriome
The gut mycobiome-bacteriome interface is significantly impacted by subsistence strategy in humans and nonhuman primates

# The gut mycobiome-bacteriome interface is significantly impacted by subsistence strategy in humans and nonhuman primates

ITS2 for fungal composition and v4 region of 16S rRNA for bacterial composition were amplified and processed through standard Qiime2 pipeline after adapter removal and quality filtering. For both bacterial and fungal quantification almost similar steps which we have used before (https://github.com/ashoks773/Oral-microbiome-16S) were followed.

The following files were used for downstream statistical analysis
1. Bacterial and fungal ASV abundances
2. Tree file (rooted)
3. Taxonomy file (using UNITE database for Fungal and GreenGenes for bacterial)
4. Metadata file

#--- All downstream analysis to generate final figures are provided in four files
1. Figure1-2_Alpha-Beta-diversity.R
2. Figure3A_and5.R
3. Figure3B.R
4. Figure4_new.R
